lib/dRep/dRepImpl.py

```diff
@@ -80,102 +77,321 @@ def run_dRep(self, ctx, params):
         """
         # ctx is the context object
         # return variables are: output
-        #BEGIN run_dRep
+ 
+        #BEGIN dereplicate
 
-        
-        dprint('ctx:', ctx)
-        dprint('params:', params)
+        dprint('params', run=locals())
 
 
+        # set up globals ds `_globals` for this API-method run
 
+        reset(_globals) # clear all fields but `debug`
```

Comment:
This is an unusual way to do things. It's confusing to both have an instance var, `self._globals`, as well as an imported var with the same name. Why both?